src/pages/AppDetails/AppDetailsPage.tsx

```diff
@@ -30,8 +38,8 @@ const emptyApp = {
   runtimes: []
 };
 
-class AppDetails extends React.Component<RouteComponentProps<AppId>, AppDetailsState> {
-  constructor(props: RouteComponentProps<AppId>) {
+class AppDetails extends React.Component<AppDetailsProps, AppDetailsState> {
+  constructor(props: AppDetailsProps) {
```

Comment:
I think now that this is sharing the duration setting in redux, it must also respect and manage the setting in the URL.  See OverviewToolbar, for example.  We typically have logic in the constructor and also componentDidUpdate,  This comment would apply to the other detail pages as well.